src/index.ts

```diff
@@ -56,6 +46,35 @@ const plugin: PluginImpl<{}> = () => {
     return { source, program };
   }
 
+  // Parse a TypeScript module into an ESTree program.
+  const references = new Set<string>();
+  function transformFile(input: ts.SourceFile): SourceDescription {
+    let code = input.getFullText();
+
+    // Collect any reference directives
+    const pragmas: PragmaMap = (input as any).pragmas;
+    if (pragmas && pragmas.has("reference")) {
```

Comment:
since you check this here, is your type declaration really correct? (nullability?)
Also, your manual type declaration plus the `as any` cast suggests that this is not officially documented TS API?

A quick check with https://astexplorer.net/#/gist/577600e28bd89938dc16f5bf445980d4/0213bb7d7515552ab575dabaec61561145080b5d showed me that the sourceFile has a `typeReferenceDirectives` array. Maybe using that would be a better idea?
(these is also `libReferenceDirectives`)

That way, you can really re-create the tripleslash references manually.
Regarding the issue you opened for `rollup`, I think it is *guaranteed* to remove those comments, right? So we will not end up with a case where we end up with duplicated comments, right?

Reply:
Nice, I didn't see the `typeReferenceDirectives` property before.



> I think it is _guaranteed_ to remove those comments, right? So we will not end up with a case where we end up with duplicated comments, right?

I don't think so. I see a preserved directive in the `.d.ts` bundle of `@react-spring/web`. 😕 
Looks like we'll need a test case for that.

Reply:
Though, that might be because I'm using the `renderChunk` hook to prepend the comment pragmas. There might be a better place to do that?

Reply:
> Looks like we'll need a test case for that.

yes, that was also my thought to test that comments that are attached to items are correctly preserved / removed. that pragma is kind of weird, according to astexplorer its attached to the export statement, but that does not really make sense IMO.